### PR TITLE
docs: Adding details about the information that `get_execution_info` syscall returns.

### PR DESCRIPTION
**Pull request by stoobie**
### Description of the Changes

This PR addresses issue #917 

### PR Preview URL

[get_execution_info](https://starknet-io.github.io/starknet-docs/pr-918/documentation/architecture_and_concepts/Smart_Contracts/system-calls-cairo1/#get_execution_info)

### Check List

- [x] Changes have been done against dev branch, and PR does not conflict
- [x] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/918)

### CODE REVIEW

File: components/Starknet/modules/architecture_and_concepts/nav.adoc [hunk 1]
```diff
@@ -31,8 +31,9 @@
 *** xref:Smart_Contracts/contract-syntax.adoc[Migrating a contract from Cairo v1 to Cairo v2]
 *** xref:Smart_Contracts/cairo-and-sierra.adoc[Cairo and Sierra]
 *** xref:Smart_Contracts/universal-deployer.adoc[Universal Deployer Contract]
-*** xref:Smart_Contracts/system-calls-cairo1.adoc[System calls]
 *** xref:Smart_Contracts/serialization_of_Cairo_types.adoc[Serialization of Cairo types]
+*** xref:Smart_Contracts/system-calls-cairo1.adoc[System calls]
+*** xref:Smart_Contracts/execution_info.adoc[Execution information]
```
Comment (stoobie) [inline components/Starknet/modules/architecture_and_concepts/nav.adoc:36]:
```suggestion
*** xref:Smart_Contracts/execution_info.adoc[Execution information for the current block]
```

**Review by ArniStarkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/918)* status: 0 of 3 files reviewed, 6 unresolved discussions (waiting on @stoobie)

___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 2 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqmOWD1Th3Ng80U-OH:-NsqmOWD1Th3Ng80U-OI:bo3fk1a) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L2)):*
> ```AsciiDoc
> [id="execution_info"]
> = Execution information for the current block
> ```

Are we sure we like this headline?

I want to say that the tx_info (and the rest of the parameters that are even more specific than the tx_info) are not the same tx info for all transactions run in a given block...
The only think that is `for the current block` in this context is the block info...



_Code quote:_
```AsciiDoc
[id="execution_info"]
= Execution information for the current block
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 11 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqmcPyAxwUhIZ-liwQ:-NsqmcPzClaQVBuLsAvl:b-w312nk) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L11)):*
> ```AsciiDoc
> 
> [horizontal,labelwidth="25",role="stripes-odd"]
> `block_info: Box<BlockInfo>`:: Contains information about a block. For details, see xref:#block_info[]
> ```

Does the link to `block_info` work here?



_Code quote:_
```AsciiDoc
`block_info: Box<BlockInfo>`:: Contains information about a block. For details, see xref:#block_info[]
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 12 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nsqm6h50YH03iC6fcpc:-Nsqm6h50YH03iC6fcpd:b4spw15) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L12)):*
> ```AsciiDoc
> [horizontal,labelwidth="25",role="stripes-odd"]
> `block_info: Box<BlockInfo>`:: Contains information about a block. For details, see xref:#block_info[]
> `tx_info: Box<TxInfo>`:: Contains information about a transaction.
> ```

add xref to tx_info.



_Suggestion:_
```AsciiDoc
`tx_info: Box<TxInfo>`:: Contains information about a transaction.  For details, see xref#tx_info[]
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 22 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqoYoV176mcizVTyMu:-NsqoYoV176mcizVTyMv:b3q4q1f) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L22)):*
> ```AsciiDoc
> [horizontal,labelwidth="25",role="stripes-odd"]
> `_block_number_: u64`:: The number of the block that is currently being executed. When called from an account contracts +`__validate__`+ function, this value is rounded down to the nearest multiple of 100.
> `_block_timestamp_: u64`:: The timestamp showing the creation time of the block. When called from an account contracts +`__validate__`+ function, this value is rounded down to the nearest hour. #Is this in UTC time? Or seconds since the Unix epoch? Or something else?#
> ```

I am not sure. I think it is UTC time...



_Code quote:_
```AsciiDoc
`_block_timestamp_: u64`:: The timestamp showing the creation time of the block. When called from an account contracts +`__validate__`+ function, this value is rounded down to the nearest hour. #Is this in UTC time? Or seconds since the Unix epoch? Or something else?#
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 36 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nsqnx4xBX2Ur0JhD2Ea:-Nsqnx4xBX2Ur0JhD2Eb:bx372fe) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L36)):*
> ```AsciiDoc
> This field can be used to prevent replay of testnet transactions on mainnet.
> `_nonce_: felt252`:: The transaction's nonce.
> `_resource_bounds_: Span<ResourceBounds>`:: A span of `ResourceBounds` structs.
> ```

Not sure I am doing it right - but I believe I have the right spirit.



_Suggestion:_
```AsciiDoc
`_resource_bounds_: Span<ResourceBounds>`:: A span of `ResourceBounds` structs. For details, see xref#resource_bounds[]
```

**Comment by ArniStarkware** [comment]
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 2 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqmOWD1Th3Ng80U-OH:-NsrS2gy7KOCPB6G9dhA:blnq3jv) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L2)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Are we sure we like this headline?

I want to say that the tx_info (and the rest of the parameters that are even more specific than the tx_info) are not the same tx info for all transactions run in a given block...
The only think that is `for the current block` in this context is the block info...
</blockquote></details>

In this context, the only thing* that is `for the current block` is the block info.

**Review by stoobie — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/918)* status: 0 of 3 files reviewed, 6 unresolved discussions (waiting on @ArniStarkware)

___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 2 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqmOWD1Th3Ng80U-OH:-NsrkGN87oAPyo4DDbuo:belxbod) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L2)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

In this context, the only thing* that is `for the current block` is the block info.
</blockquote></details>

OK. I changed to "Block execution info" WDYT?
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 11 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqmcPyAxwUhIZ-liwQ:-NsrkWlZCDNd8pfiisT4:bh46egj) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L11)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Does the link to `block_info` work here?
</blockquote></details>

Done. Fixed.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 12 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nsqm6h50YH03iC6fcpc:-NsrkiVuC4NgVQkRfNCV:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L12)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

add xref to tx_info.
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 22 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqoYoV176mcizVTyMu:-NsrkoZI8j1ePkVHYM8x:b-fehuz5) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L22)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

I am not sure. I think it is UTC time...
</blockquote></details>

Who would know the answer?
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 36 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nsqnx4xBX2Ur0JhD2Ea:-NsrlhZ5A3r-dFUoGbaC:b-el33eg) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L36)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Not sure I am doing it right - but I believe I have the right spirit.
</blockquote></details>

Done. :)

**Review by ArniStarkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/918)* status: 0 of 3 files reviewed, 7 unresolved discussions (waiting on @stoobie)

___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 2 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqmOWD1Th3Ng80U-OH:-NsryWj848B8UBLlH5Eq:bqj9qjv) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L2)):*
<details><summary><i>Previously, stoobie (Steve Goodman) wrote…</i></summary><blockquote>

OK. I changed to "Block execution info" WDYT?
</blockquote></details>

How about `Execution information`? No specific refer to the `block`.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 22 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsrzsgsBMLG3tmQXYhI:-NsrzsgsBMLG3tmQXYhJ:b71gd84) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L22)):*
> ```AsciiDoc
> 
> [horizontal,labelwidth="25",role="stripes-odd"]
> `_block_number_: u64`:: The number of the block that is currently being executed. When called from an account contracts +`__validate__`+ function, this value is rounded down to the nearest multiple of 100.
> ```

true for any `__validate__` type function. See the comment below. 



_Code quote:_
```AsciiDoc
When called from an account contracts +`__validate__`+ function,
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 23 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nss-6tR7Ypsi3ZdEPrG:-Nss-6tR7Ypsi3ZdEPrH:b71gd84) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L23)):*
> ```AsciiDoc
> [horizontal,labelwidth="25",role="stripes-odd"]
> `_block_number_: u64`:: The number of the block that is currently being executed. When called from an account contracts +`__validate__`+ function, this value is rounded down to the nearest multiple of 100.
> `_block_timestamp_: u64`:: The timestamp showing the creation time of the block, in seconds since the Unix epoch, based on UTC time, rounded down to the nearest second. When called from an account contract's +`__validate__`+ function, this value is rounded down to the nearest hour.
> ```

true for any `__validate__` type function. See the comment below. 



_Code quote:_
```AsciiDoc
When called from an account contract's +`__validate__`+ function, this value is rounded down to the nearest hour.
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 30 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nss-FjTDbenNJwW7Z7V:-Nss-FjTDbenNJwW7Z7W:undefined) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L30)):*
> ```AsciiDoc
> 
> [horizontal,labelwidth="25",role="stripes-odd"]
> `_version_: felt252`:: The version of the transaction. It is fixed (currently, 1) in the OS, and should be signed by the account contract. This field allows invalidating old transactions, whenever the meaning of the other transaction fields is changed (in the OS).
> ```





_Suggestion:_
```AsciiDoc
`_version_: felt252`:: The version of the transaction. It is fixed (currently, 3) in the OS, and should be signed by the account contract. This field allows invalidating old transactions, whenever the meaning of the other transaction fields is changed (in the OS).
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc` line 68 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nsrv3CMEEeWIA67239u:-Nsrv3CMEEeWIA67239v:blipqci) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc#L68)):*
> ```AsciiDoc
> In Cairo 1.0, this single system call contains all information for a block, transaction, and execution context.
> 
> When an account's `+__validate__+` function calls `get_execution_info`:
> ```

I am not sure if this is a technical fix.

This statement is true for any "`__validate__` type" function.
I mean to say this is true for either of the functions: `__validate__`, `__validate_declare__`, `__validate_deploy__`.



_Code quote:_
```AsciiDoc
When an account's `+__validate__+` function calls `get_execution_info`:
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc` line 71 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsrxZ8q0PuAw24yBGgP:-NsrxZ8q0PuAw24yBGgQ:b-89pips) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc#L71)):*
> ```AsciiDoc
> 
> * `block_timestamp` returns the hour, rounded down to the nearest hour.
> * `block_number` returns the block number, rounded down to the nearest multiple of 100.
> ```

A different topic than my last comment:
Here is a suggestion.

It is odd to state what the syscall returns in validate mode but not explain what is returned in any other case.

To see what is usually returned by this syscall, one needs to look into the other file first. maybe link it here as well?
I like that this information is present here. It will help any confused developer dealing with this syscall (dealing with odd bugs related to validate).



_Suggestion:_
```AsciiDoc
When an account's `+__validate__+` function calls `get_execution_info` some of the returned values are modified:

* `block_timestamp` returns the hour, rounded down to the nearest hour.
* `block_number` returns the block number, rounded down to the nearest multiple of 100.
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc` line 82 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsrypUiDfCPBdQUiD_R:-NsrypUiDfCPBdQUiD_S:byvmijb) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc#L82)):*
> ```AsciiDoc
> 
> [horizontal,labelwidth="25",role="stripes-odd"]
> link:https://github.com/starkware-libs/cairo/blob/main/corelib/src/starknet/info.cairo#L8[`ExecutionInfo`]:: A struct that contains information about the currently executing block and the transactions in the block.
> ```

The information is related to either:
The currently executed function.
The currently executed transaction.
The currently executed block.

all of them are tightly connected to the current run. Non of the info is referring to other transactions.



_Suggestion:_
```AsciiDoc
link:https://github.com/starkware-libs/cairo/blob/main/corelib/src/starknet/info.cairo#L8[`ExecutionInfo`]:: A struct that contains information about the currently executing function, transaction and block.
```

**Review by stoobie — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/918)* status: 0 of 3 files reviewed, 7 unresolved discussions (waiting on @ArniStarkware)

___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 2 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsqmOWD1Th3Ng80U-OH:-Nss5M3_79XBuXf5Hwi_:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/207c61252fcb983be4b48eeaa801ef351b84532f/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L2)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

How about `Execution information`? No specific refer to the `block`.
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 22 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsrzsgsBMLG3tmQXYhI:-NssFXKZ5j-vGMBoKXSo:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L22)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

true for any `__validate__` type function. See the comment below.
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 23 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nss-6tR7Ypsi3ZdEPrG:-NssFY1ZD6_CZxfzeqfr:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L23)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

true for any `__validate__` type function. See the comment below.
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 30 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nss-FjTDbenNJwW7Z7V:-Nss7vaHE-cTY0pYQAwa:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L30)):*
> ```AsciiDoc
> 
> [horizontal,labelwidth="25",role="stripes-odd"]
> `_version_: felt252`:: The version of the transaction. It is fixed (currently, 1) in the OS, and should be signed by the account contract. This field allows invalidating old transactions, whenever the meaning of the other transaction fields is changed (in the OS).
> ```

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc` line 68 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nsrv3CMEEeWIA67239u:-Nss8LX65yWjx5Nm-Y4B:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc#L68)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

I am not sure if this is a technical fix.

This statement is true for any "`__validate__` type" function.
I mean to say this is true for either of the functions: `__validate__`, `__validate_declare__`, `__validate_deploy__`.
</blockquote></details>

Done.
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc` line 71 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsrxZ8q0PuAw24yBGgP:-NssEJy-3jFTETDEv9zb:b-ky3j8c) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc#L71)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

A different topic than my last comment:
Here is a suggestion.

It is odd to state what the syscall returns in validate mode but not explain what is returned in any other case.

To see what is usually returned by this syscall, one needs to look into the other file first. maybe link it here as well?
I like that this information is present here. It will help any confused developer dealing with this syscall (dealing with odd bugs related to validate).
</blockquote></details>

I moved the link to the other page to the top of this section. How's that?

I changed the text to:
```
When an account's `+__validate__+`, +`__validate_deploy__`+, or +`__validate_declare__`+ function calls `get_execution_info`, the return values for `block_timestamp` and `block_number` are modified as follows:
```
___
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc` line 82 at r4](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsrypUiDfCPBdQUiD_R:-NssF62f7Cl2Cvm-M78L:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/7ee91e3ce0b859a0f1647498742f4c20ad92f427/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc#L82)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

The information is related to either:
The currently executed function.
The currently executed transaction.
The currently executed block.

all of them are tightly connected to the current run. Non of the info is referring to other transactions.
</blockquote></details>

Done.

**Review by ArniStarkware — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r3, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-:-NsskGpH9LLyVGhUWVvW:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @stoobie)

**Comment by ArniStarkware** [comment]
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 22 at r5](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nssl3aZDEgrsj_t0NPF:-Nssl3a_4uegeUY1deNz:undefined) ([raw file](https://github.com/starknet-io/starknet-docs/blob/83bcf951e4b339a98b9e4a3acaf841d9de783588/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L22)):*
> ```AsciiDoc
> 
> [horizontal,labelwidth="25",role="stripes-odd"]
> `_block_number_: u64`:: The number of the block that is currently being executed. When called from an account contract's +`__validate__`+, +`__validate_deploy__`+, or +`__validate_declare__`+ function, this value is rounded down to the nearest multiple of 100.
> ```





_Suggestion:_
```AsciiDoc
`_block_number_: u64`:: The number of the block that is currently being executed. When called from an account contract's +`__validate__`+, +`__validate_deploy__`+, +`__validate_declare__`+, or +`constructor`+ function, this value is rounded down to the nearest multiple of 100.
```

**Comment by ArniStarkware** [comment]
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc` line 23 at r5](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-Nssktqk2fm6OrgQNH55:-Nssktqk2fm6OrgQNH56:undefined) ([raw file](https://github.com/starknet-io/starknet-docs/blob/83bcf951e4b339a98b9e4a3acaf841d9de783588/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/execution_info.adoc#L23)):*
> ```AsciiDoc
> [horizontal,labelwidth="25",role="stripes-odd"]
> `_block_number_: u64`:: The number of the block that is currently being executed. When called from an account contract's +`__validate__`+, +`__validate_deploy__`+, or +`__validate_declare__`+ function, this value is rounded down to the nearest multiple of 100.
> `_block_timestamp_: u64`:: The timestamp showing the creation time of the block, in seconds since the Unix epoch, based on UTC time, rounded down to the nearest second. When called from an account contract's +`__validate__`+, +`__validate_deploy__`+, or +`__validate_declare__`+ function, this value is rounded down to the nearest hour.
> ```





_Suggestion:_
```AsciiDoc
When called from an account contract's +`__validate__`+, +`__validate_deploy__`+, +`__validate_declare__`+, or +`constructor`+ function, this value is rounded down to the nearest hour.
```

**Comment by ArniStarkware** [comment]
*[`components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc` line 68 at r5](https://reviewable.io/reviews/starknet-io/starknet-docs/918#-NsslK_VA5eLJEahytRw:-NsslK_VA5eLJEahytRx:undefined) ([raw file](https://github.com/starknet-io/starknet-docs/blob/83bcf951e4b339a98b9e4a3acaf841d9de783588/components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/system-calls-cairo1.adoc#L68)):*
> ```AsciiDoc
> This single system call contains all information for a block, transaction, and execution context.
> 
> When an account's `+__validate__+`, +`__validate_deploy__`+, or +`__validate_declare__`+ function calls `get_execution_info`, the return values for `block_timestamp` and `block_number` are modified as follows:
> ```





_Suggestion:_
```AsciiDoc
When an account's `+__validate__+`, +`__validate_deploy__`+, +`__validate_declare__`+, or +`constructor`+ function calls `get_execution_info`, the return values for `block_timestamp` and `block_number` are modified as follows:
```

**Pull request merged**